### ISSUE TITLE
Augment getTransaction response to include events emitted during the transaction

### ISSUE THREAD

**Issue by tamirms** [stellar/soroban-tools#1041]
### What problem does your feature solve?

There is no easy way to get events for a transaction. If you have a failed transaction in particular, it would be nice to be able to debug it by viewing the diagnostic events that were emitted during the transaction.

### What would you like to see?

Include events in the `getTransaction` response so we can see what events where emitted during the transaction.

### What alternatives are there?

Manually decoding the tx meta xdr in the `getTransaction` response and then extract the events from the tx meta.

You could also find events for a tx using two rpc calls:

1. Call `getTransaction` to get the ledger and tx index for the transaction in question.
2. Call `getEvents` with the cursor set appropriately so that includes the ledger and tx index

**Comment by mollykarcher** [comment]
We'd like to keep this within `getEvents`, rather than `getTransaction` if possible. We should investigate if it's possible to do this in a single call when adding additional indices.

@tyvdh can you give us some context on how people are typically attempting to use this?

**Comment by 2opremio** [comment]
> You could also find events for a tx using two rpc calls:
> 
>     1. Call `getTransaction` to get the ledger and tx index for the transaction in question.
> 
>     2. Call `getEvents` with the cursor set appropriately so that includes the ledger and tx index

I would advocate for this. We can make it a single call if we keep an in-memory index of txHash -> Ledger+tx Index (which shouldn't be too hard)

**Comment by tamirms** [comment]
It would be easier to include the events in the `getTransaction` response because we already have the tx meta for the transaction stored in-memory. We could implement this feature by parsing out the events from the tx meta in the `getTransaction` handler. We wouldn't need to store any additional data in the transaction or event memory stores.

**Comment by tyvdh** [comment]
I'm actually not sure how painful of an issue this is. With recent improvements to the JS SDK I've been using:

> Manually decoding the tx meta xdr in the getTransaction response and then extract the events from the tx meta.

And that works fine. There are other things the [getEvents](https://github.com/stellar/soroban-tools/issues/842) and [getTransaction](https://github.com/stellar/soroban-tools/issues/1056) endpoints need but at least for me and the Discord conversations I'm seeing this isn't a pain point.

**Comment by leighmcculloch** [comment]
> It would be easier to include the events in the `getTransaction` response because we already have the tx meta for the transaction stored in-memory.

@tamirms Could the getEvents endpoint when a tx id is passed be backed by the getTransaction?

**Comment by tamirms** [comment]
@leighmcculloch you are right, we could allow the getEvents endpoint to be called with a tx id. The implementation should be pretty simple as we can lookup the transaction by hash using the in-memory transaction store that backs the getTransaction endpoint. Once we have the transaction, we can decode the events from the tx meta in the transaction

**Comment by psheth9** [comment]
Posting this questions from my limited understanding :) 

1. Based on discussion it seems like we want to update `getEvents` request/response and not the getTxn response.
If that is the case, how do we know the required fields of getEvents endpoint e.g `startLedger` and `filter`? Is it okay to make it optional or will users have all those details while calling?

2. It looks like users use getTxn endpoint in order to check the status of specific transaction, It would be more useful to just amend response of getTxn endpoint otherwise users need to call two endpoints in order to find out what went wrong (getTxn and then getEvents)

**Comment by tomerweller** [comment]
I think that this issue got side tracked a bit with `getEvents`. I support @tamirms original ask of including events in the result of `getTransaction`. 

Events are important in the context of the `getTransaction` result. Clients are already polling that endpoint for success/failure and when they get it inspecting for the events emitted makes sense. We know that developers are already doing this but they are forced to decode metas for it, which is not great in terms of ergonomics. Asking the developer to make another request to get that information is also not great ergonomics. 

From an implementation perspective do we really need to use the events backend? Why not decode events from metas on the fly when handling requests?

**Comment by 2opremio** [comment]
> From an implementation perspective do we really need to use the events backend? Why not decode events from metas on the fly when handling these requests?

Yep, we can do this.
